libkineto/src/CuptiActivityProfiler.h

```diff
@@ -453,6 +453,8 @@ class CuptiActivityProfiler {
   // Mutex to protect non-atomic access to below state
   std::mutex mutex_;
 
+  std::thread * stopByIterThread = nullptr;
```

Comment:
can we make this a std::unique_ptr so its management is clean RAII

Reply:
please add a comment that this thread is used to asynchronously trace stoping when used by iteration based profiling